### ISSUE TITLE
IPv6/multihome support

### ISSUE THREAD

**Issue by ssiloti** [arvidn/libsimulator#1]
I'd like to use this to test dual stack `dht_tracker`s in libtorrent, but to do that I need IPv6 and multihome support so that a single io_service can have both IPv4 and IPv6 addresses. I'm currently scoping out what's going to be required to implement this.

**Comment by arvidn** [comment]
oh, I just saw this. I was working on multi-home support independently, partly to support and old outstanding patch I've been working on for libtorrent to clean up the logic around creating listening sockets. But also to be able to test and fix the tracker announce issue on IPv6 reported against libtorrent.

I believe the multi-home support is complete, and I managed to fix the tracker announce IPv6 issue.

just pass in multiple IP addresses to the sim::asio::io_service constructor. Just keep in mind that address_v4::any() will bind to _just_ the first IPv4 address, and address_v6::any() will bind only to the first IPv6 address. i.e. there's no binding a single socket to multiple interfaces. I don't think libtorrent relies on being able to do that though.
